File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.py
# 그냥 다 조합해보고 중복제거, 정렬
# 시간복잡도 약 O(n^2)

def solution(numbers):
    answer = []
    numbers.sort()
    result = []
    
    length = len(numbers)
    for i in range(length - 1):
        for j in range(i+1, length) :
            result.append(numbers[i] + numbers[j])    
    result = set(result)
    result = list(result)
    result.sort()
    return result

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        HashSet<Integer> numSet = new HashSet<>();
        int length = numbers.length;
        for(int i=0; i<length; i++) {
            for(int j=i+1; j<length; j++) {
                numSet.add(numbers[i] + numbers[j]);
            }
        }
        
        return numSet.stream()
            .sorted()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}